clients/src/main/java/org/apache/kafka/clients/consumer/internals/CommitRequestManager.java

```diff
@@ -209,15 +218,21 @@ public NetworkClientDelegate.UnsentRequest toUnsentRequest() {
                 requestTopicDataMap.put(topicPartition.topic(), topic);
             }
 
-            OffsetCommitRequest.Builder builder = new OffsetCommitRequest.Builder(
-                    new OffsetCommitRequestData()
-                            .setGroupId(this.groupId)
-                            .setGenerationIdOrMemberEpoch(generation.generationId)
-                            .setMemberId(generation.memberId)
-                            .setGroupInstanceId(groupInstanceId)
-                            .setTopics(new ArrayList<>(requestTopicDataMap.values())));
+            OffsetCommitRequestData data = new OffsetCommitRequestData()
+                    .setGroupId(groupId)
+                    .setGenerationIdOrMemberEpoch(memberEpoch)
```

Comment:
Aligned with the idea that memberEpoch may not be present, I expect we should only set this if we have one (like it is done for the memberID)

Reply:
Agreed. Updated.